static/js/biomedical/landing/biomedical_search_box.tsx

```diff
@@ -0,0 +1,135 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Search box for the Biomedical DC landing page
+ */
+
+import React, { useEffect, useRef, useState } from "react";
+import { Input, InputGroup } from "reactstrap";
+import { styled } from "styled-components";
+
+import { BREAKPOINTS } from "./shared";
+
+const SearchIcon = styled.div`
+  align-items: center;
+  display: flex;
+  margin-left: 4px;
+  padding: 8px;
+`;
+
+const StyledInputGroup = styled(InputGroup)`
+  background-color: white;
+  border-radius: 28px;
+  border: 1px solid #303030;
+  display: flex;
+  margin: 48px 0px 100px 0px;
+  padding: 8px 0px;
+
+  &:focus-within {
+    outline: 3px solid ${(props) => props.theme.highlightColors.main};
+  }
+
+  @media ${BREAKPOINTS.md} {
+    margin: 22px 0px 48px 0px;
+  }
+`;
+
+const StyledInput = styled(Input)`
+  border: none;
+  color: ${(props) => props.theme.header.textColorLight};
+  font-size: 16px;
+  font-weight: 500;
+  line-height: 24px;
+  margin-right: 20px;
+  overflow: hidden;
+  resize: none;
+
+  @media ${BREAKPOINTS.md} {
+    font-size: 14px;
+    font-weight: 400;
+    line-height: 18px;
+  }
+
+  &:focus {
+    box-shadow: none !important;
+  }
+`;
+
+interface BiomedicalSearchProps {
+  // Function to run on search
+  onSearch: (query: string) => void;
+  // Placeholder query to show
+  placeholderText?: string;
+}
+
+export function BiomedicalSearchBox(props: BiomedicalSearchProps): JSX.Element {
+  const [invalid, setInvalid] = useState(false);
+  const [value, setValue] = useState<string>("");
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  // Dynamically resize search box height to show entire query
+  const setSearchBoxHeight = () => {
+    inputRef.current.style.height = "inherit";
+    inputRef.current.style.height = `${inputRef.current.scrollHeight}px`;
+  };
+
+  // Adjust search box height when window is resized
+  useEffect(() => {
+    window.addEventListener("resize", setSearchBoxHeight);
+  });
+
+  // Adjust search box height while query is being entered
+  useEffect(() => {
+    setSearchBoxHeight();
+  }, [props.placeholderText, value]);
+
+  return (
+    <StyledInputGroup>
+      <SearchIcon>
+        <span className="material-icons-outlined">search</span>
+      </SearchIcon>
+      <StyledInput
+        autoComplete="off"
+        autoFocus={false}
+        innerRef={inputRef}
+        invalid={invalid}
```

Comment:
Hitting enter with no query gives me this: https://screenshot.googleplex.com/3F6GPMBD47jonjz

Reply:
Good catch! Looks like that's from Reactstrap's default invalid behavior. We don't do anything with the "invalid" state, so I've set invalid to false instead.